acstools/findsat_mrt.py

```diff
@@ -0,0 +1,1129 @@
+        output_dir : string, optional
+            Path in which to save output. The default is './'.
+        output_root : string, optional
+            A prefix for all output files. The default is ''.
+        check_persistence : bool, optional
+            Calculates the persistence of all identified streaks. The default is True.
+        min_persistence : float, optional
+            Minimum persistence of a "true" satellite trail. Must be between 0 and 1. The default is 0.5. Note that this does not reject satellite trails from the output catalog, but highlights them in a different color in the output plot.
+        ignore_theta_range : array-like, optional
+            List if ranges in theta to ignore when identifying satellite trails. This parameter is most useful for avoiding false positives due to diffraction spikes
+            that always create streaks around the same angle for a given telescope/instrument. Format should be a list of tuples, e.g., [(theta0_a,theta1_a),(theta0_b,theta1_b)].
+            Default is None.
+        save_catalog: bool, optional
+            Set to save the catalog of identified trails. Default is True
+        save_diagnostic: bool, optional
+            Set to save a diagnotic plot showing the input image and identified trails. Default is True.
+        save_mrt: bool, optional
+            Set to save the MRT in a fits file. Default is false.
+        save_mask: bool, optional
+            Set to save the trail mask in a fits file. Default is false.
+
+        Returns
+        -------
+        None.
+
+        '''
+
+        #inputs
+        self.image=image
+        self.header=header
+        self.image_header=image_header
+        self.save_image_header_keys = save_image_header_keys
+        self.threshold=threshold
+        self.min_length=min_length
+        self.max_width=max_width
+        self.kernels=kernels
+        self.threads=threads
+        self.theta=theta
+        self.plot=plot
+        self.buffer=buffer
+        self.check_persistence=check_persistence
+        self.min_persistence=min_persistence
+        self.ignore_theta_range = ignore_theta_range
+                
+        #outputs
+        self.mrt=None
+        self.mrt_err=None
+        self.length=None
+        self.rho=None
+        self.mask=None
+        
+        #some internal things
+        self._madrt=None
+        self._medrt=None
+        self._image_mad=None
+        self._image_stddev=None
+
+        #info for saving output
+        self.output_dir = output_dir
+        self.root = output_root
+        self.save_catalog = save_catalog
+        self.save_diagnostic = save_diagnostic
+        self.save_mrt = save_mrt
+        self.save_mask = save_mask
+        
+        #plot image upon initialization
+        if (np.any(image) != None) & (self.plot == True):
+            self.plot_image()
+
+    def run_mrt(self,theta=None,threads=None):
```

Comment:
The idea here was flexibility and it being clear which attributes from trailfinder go into which functions. For instance, without the theta and threads keywords, it may not be obvious that these are things that can be adjusted. Is there a best practice here? If possible, I'd like to keep the keyword options, but if they end up being used, I could then update the relevant attributes of the satfinder class.

Reply:
I searched around a bit, but couldn't find a specific best practice for this kind of thing. I'd be fine with keeping the keywords and updating the attributes if they are used.